File: CS/WindowsApplication7/MenuHelper.cs
using System.Drawing;
using System.Windows.Forms;
using DevExpress.Utils.Menu;
using DevExpress.XtraTab.ViewInfo;
using DevExpress.XtraTabbedMdi;

namespace WindowsApplication7 {
    public class MenuHelper {
        DXPopupMenu menuCore;
        XtraTabbedMdiManager managerCore;
        public event PopupMenuShowingEventHandler PopupMenuShowing;

        public MenuHelper(XtraTabbedMdiManager manager) {
            this.managerCore = manager;
            this.managerCore.MouseUp += OnManagerMouseUp;
        }
        public DXPopupMenu Menu {
            get {
                if(menuCore == null)
                    menuCore = new DXPopupMenu();
                return menuCore;
            }
            set { menuCore = value; }
        }
        void OnManagerMouseUp(object sender, MouseEventArgs e) {
            if(e.Button != MouseButtons.Right)
                return;
            BaseTabHitInfo hi = managerCore.CalcHitInfo(new Point(e.X, e.Y));
            if(hi.HitTest == XtraTabHitTest.PageHeader) {
                Menu.Items.Clear();
                PopupMenuShowingEventArgs args = new PopupMenuShowingEventArgs(hi.Page) { Menu = Menu };
                RaisePopupMenuShowing(args);
                ((IDXDropDownControl)Menu).Show(new SkinMenuManager(DevExpress.LookAndFeel.UserLookAndFeel.Default.ActiveLookAndFeel),
                   managerCore.MdiParent.ActiveMdiChild,
                   e.Location);

            }
        }
        void RaisePopupMenuShowing(PopupMenuShowingEventArgs e) {
            if(PopupMenuShowing != null)
                PopupMenuShowing(managerCore, e);
        }
    }
}

File: VB/WindowsApplication7/MenuHelper.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraTab.ViewInfo
Imports DevExpress.XtraTabbedMdi

Namespace WindowsApplication7
	Public Class MenuHelper
		Private menuCore As DXPopupMenu
		Private managerCore As XtraTabbedMdiManager
		Public Event PopupMenuShowing As PopupMenuShowingEventHandler

		Public Sub New(ByVal manager As XtraTabbedMdiManager)
			Me.managerCore = manager
			AddHandler Me.managerCore.MouseUp, AddressOf OnManagerMouseUp
		End Sub
		Public Property Menu() As DXPopupMenu
			Get
				If menuCore Is Nothing Then
					menuCore = New DXPopupMenu()
				End If
				Return menuCore
			End Get
			Set(ByVal value As DXPopupMenu)
				menuCore = value
			End Set
		End Property
		Private Sub OnManagerMouseUp(ByVal sender As Object, ByVal e As MouseEventArgs)
			If e.Button <> MouseButtons.Right Then
				Return
			End If
			Dim hi As BaseTabHitInfo = managerCore.CalcHitInfo(New Point(e.X, e.Y))
			If hi.HitTest = XtraTabHitTest.PageHeader Then
				Menu.Items.Clear()
				Dim args As New PopupMenuShowingEventArgs(hi.Page) With {.Menu = Menu}
				RaisePopupMenuShowing(args)
				CType(Menu, IDXDropDownControl).Show(New SkinMenuManager(DevExpress.LookAndFeel.UserLookAndFeel.Default.ActiveLookAndFeel), managerCore.MdiParent.ActiveMdiChild, e.Location)

			End If
		End Sub
		Private Sub RaisePopupMenuShowing(ByVal e As PopupMenuShowingEventArgs)
			RaiseEvent PopupMenuShowing(managerCore, e)
		End Sub
	End Class
End Namespace
